### ISSUE TITLE
`--no-inline` Feature no longer working.

### ISSUE THREAD

**Issue by Bromilow** [HaxeFoundation/haxe#10281]
We are having issues with our inline functions since upgrading. Where the `--no-inline` flag does not seem to be working anymore.

I can confirm the `--no-inline` feature was working in version `4.1.5`, but stopped working from version  `4.2.0`

Functions that are inlined now fail with the error: 
`Field {name} is inlined and cannot be overridden`

Version 4.2.0 (error)
![example](https://user-images.githubusercontent.com/16894617/120313229-fd66da00-c2d9-11eb-83fd-93a4a025fa72.PNG) version 

Version 4.1.5 (working)
![example2](https://user-images.githubusercontent.com/16894617/120314774-e0330b00-c2db-11eb-97f3-ccf1dca4a786.PNG)

Appreciate the help

**Comment by Simn** [comment]
As mentioned on slack: `override` and `inline` cannot be combined because they are conflicting concepts. Overriding leads to dynamic dispatch, which means method selection at run-time, whereas inlining is a compile-time feature that requires the compiler to know which method it is supposed to inline.

The reason this used to work is that previously, `--no-inline` would work in a way that essentially removed the `inline` flag from the method, which resolved the conflict. This incidentally allowed the code to compile, but that's not a good thing: The declaration of of that `override inline function` is unsound and should always be caught by the compiler, no matter which compilation flags are used.

In summary, you can override or you can inline, but you cannot do both. On our end, this now works as expected. Thank you for the report!
